pkg/controller/installation/products/launcher/reconciler.go

```diff
@@ -215,136 +222,140 @@ func (r *Reconciler) reconcileRHSSOIntegration(ctx context.Context, serverClient
 	}
 
 	rhssoNamespace := rhssoConfig.GetNamespace()
-	rhssoRealm := rhssoConfig.GetRealm()
-	if rhssoNamespace == "" || rhssoRealm == "" {
-		logrus.Info("Cannot configure SSO integration without SSO namespace and SSO realm")
-		return v1alpha1.PhaseInProgress, nil
+
+	kcClient := &keycloak.KeycloakClient{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("integreatly-%v-client", clientId),
+			Namespace: rhssoNamespace,
+		},
 	}
 
-	kcr := &aerogearv1.KeycloakRealm{}
-	err = serverClient.Get(ctx, pkgclient.ObjectKey{Name: rhssoRealm, Namespace: rhssoNamespace}, kcr)
+	or, err := controllerutil.CreateOrUpdate(ctx, serverClient, kcClient, func(existing runtime.Object) error {
+		client := existing.(*keycloak.KeycloakClient)
+		client.Spec = getKeycloakClientSpec(launcherUrl)
+		return nil
+	})
+
 	if err != nil {
-		return v1alpha1.PhaseFailed, err
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "could not create/update codeready keycloak client")
 	}
 
-	if !aerogearv1.ContainsClient(kcr.Spec.Clients, clientId) {
-		logrus.Infof("Adding keycloak realm client")
-		kcr.Spec.Clients = append(kcr.Spec.Clients, &aerogearv1.KeycloakClient{
-			KeycloakApiClient: &aerogearv1.KeycloakApiClient{
-				ID:                      clientId,
-				ClientID:                clientId,
-				ClientAuthenticatorType: "client-secret",
-				Enabled:                 true,
-				PublicClient:            true,
-				RedirectUris: []string{
-					launcherUrl,
-					launcherUrl + "/*",
-				},
-				WebOrigins: []string{
-					launcherUrl,
-					launcherUrl + "/*",
-				},
-				StandardFlowEnabled: true,
-				RootURL:             launcherUrl,
-				FullScopeAllowed:    true,
-				Access: map[string]bool{
-					"view":      true,
-					"configure": true,
-					"manage":    true,
-				},
-				ProtocolMappers: []aerogearv1.KeycloakProtocolMapper{
-					{
-						Name:            "given name",
-						Protocol:        "openid-connect",
-						ProtocolMapper:  "oidc-usermodel-property-mapper",
-						ConsentRequired: true,
-						ConsentText:     "${givenName}",
-						Config: map[string]string{
-							"userinfo.token.claim": "true",
-							"user.attribute":       "firstName",
-							"id.token.claim":       "true",
-							"access.token.claim":   "true",
-							"claim.name":           "given_name",
-							"jsonType.label":       "String",
-						},
+	r.logger.Infof("The operation result for keycloakclient %s was %s", kcClient.Name, or)
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func getKeycloakClientSpec(launcherUrl string) keycloak.KeycloakClientSpec {
+	return keycloak.KeycloakClientSpec{
+		RealmSelector: &metav1.LabelSelector{
+			MatchLabels: rhsso.GetInstanceLabels(),
```

Comment:
Which client is this being added too? Is it the same as realm for codeready?

Reply:
It's added to the launcher client: https://github.com/integr8ly/integreatly-operator/pull/202/files/04c2082ee23977ddf8ca3b5b7e8107cf9d6b00e3#diff-a52ceb0461f0b7d9c06ce3093fdd81a1R235